.github/workflows/ci-static-analysis.yml

```diff
@@ -0,0 +1,42 @@
+name: CI static analysis
+
+on: pull_request
+
+jobs:
+  tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        python-version: ['3.8']
+        os: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install system requirements
+        run: sudo apt install -y git libffi-dev libmysqlclient-dev libssl-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev libxmlsec1-dev liblzma-dev software-properties-common tk-dev xz-utils zlib1g-dev
```

Comment:
do we really need all these packages just to run static analysis, I've recently discovered #28539 we don't need most of these packages for CI

Reply:
You are absolutely right; I'm quite surprised that the Github worker does not seem to need all these packages, but it seems that we can run static analysis with just the libxmlsec1-dev package (and its dependencies).